### PR TITLE
[IMP] website_event: front promenade

### PR DESCRIPTION
**Pull request by Odoonan**
1. Add country flag when filter option is on
2. Reduce search bar width for >lg screens
3. Show tags inside the category drop down title instead of in a separate line, for larger screen. Also add a 'All' option to clear all searched tags of that category.

Task-3869922

### CODE REVIEW

**Review by reth-odoo — COMMENTED** [review]
Hey :wave: 

A few design suggestions, note about incoming conflict and and a couple tiny code suggestions.

Feel free to green the task once addressed :)

EDIT: sidenote, seems to be a comment from Testing India about adding a cross to delete, not sure if it was discussed. Seems ok like this imo.

File: addons/website_event/static/src/scss/event_templates_list.scss [hunk 1]
```diff
@@ -40,6 +40,19 @@
             padding: ($spacer * .5) $card-spacer-x;
             transform: translateY(-50%);
         }
+        .o_wevent_event_country_flag {
+            top: $card-spacer-x + 1rem;
+            right: $card-spacer-x + 5rem;
+            height: 2rem;
+            width: 2rem;
+            border-radius: 50%;
```
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/website_event/static/src/scss/event_templates_list.scss:48]: Yes, so I went for a default shape with height auto.

Fun fact, all flags but two fall in the 1 -> 2 ratio, then qatar is 2.5 and nepal is about 0.82.
So I used a width that made them all look fine (and centered them ofc)

_fun with flags!_

File: addons/website_event/views/event_templates_list.xml [hunk 2]
```diff
@@ -51,10 +51,15 @@
...
+                            <t t-set="searched_category_tags" t-value="category.tag_ids.filtered(lambda tag: tag.color and tag in search_tags)"/>
                             <a href="#" role="button" class="btn btn-light dropdown-toggle" data-bs-toggle="dropdown">
-                                <t t-out="category.name"/>
+                                <t t-out="', '.join(searched_category_tags.mapped('name')) or category.name"/>
                             </a>
                             <div class="dropdown-menu">
+                                <a t-att-href="'/event?%s' % keep_query('*', tags=str((search_tags - searched_category_tags).ids))"
+                                   t-attf-class="dropdown-item d-flex align-items-center justify-content-between #{'active' if not searched_category_tags else ''}">
+                                   All
+                                </a>
```
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/website_event/views/event_templates_list.xml:62]: TODO

File: addons/website_event/views/event_templates_list.xml [hunk 3]
```diff
@@ -51,10 +51,15 @@
                 <h2 class="h4 my-0 me-auto pe-sm-4">Events</h2>
                     <t t-foreach="categories" t-as="category">
                         <div t-if="category.is_published and category.tag_ids and any(tag.color for tag in category.tag_ids)" class="dropdown d-none d-lg-block">
+                            <t t-set="searched_category_tags" t-value="category.tag_ids.filtered(lambda tag: tag.color and tag in search_tags)"/>
```
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/website_event/views/event_templates_list.xml:54]: Mmh, how can we use a correct default value if category not in the search tags ? We could loop on the search tags grouped by category but it won't work as we must loop all categories in any case.

```xml
<t t-set="searched_tags_per_category" t-value="search_tags.grouped('category_id')">
...
    <t t-set="searched_category_tags" t-value="searched_tags_per_category.get(category, ???)"/>
```

However I see in `_search_get_detail` that the check on color and on tag existence is already done there so this line could probably be simplified to

```html
<t t-set="searched_category_tags" t-value="search_tags.filtered(lambda tag: tag.category = category)"/>
```

what do you think?

**Comment by Odoonan** [comment]
Hello @reth-odoo thanks for the review, left small comments, when you have time. I will still green it as suggested, as the discussion will probably not impact testing. Cheers!

File: addons/website_event/views/event_templates_list.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (reth-odoo) [reply addons/website_event/views/event_templates_list.xml]: Yep, goal was just to be more straightforward so this works for me.

Would have been valid to use `request.env['event.tag']` as default I think though

File: addons/website_event/static/src/scss/event_templates_list.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (reth-odoo) [reply addons/website_event/static/src/scss/event_templates_list.scss]: Vexillologists around the world rejoice! 🇳🇵

**Pull request closed without merging**